core/src/main/scala/org/apache/spark/sql/SnappySession.scala

```diff
@@ -1889,9 +1889,13 @@ object SnappySession extends Logging {
     def apply(session: SnappySession, lp: LogicalPlan, sqlText: String, pls:
     ArrayBuffer[ParamLiteral]): CachedKey = {
 
+      val invalidate = Array(false)
```

Comment:
No var was intentionally avoided. Using val is the recommended way. It is like final in java is preferred wherever applicable.